src/libNetwork/DataSender.cpp

```diff
@@ -237,16 +237,18 @@ bool DataSender::SendDataToOthers(
     unsigned int my_shards_lo = 0;
     unsigned int my_shards_hi = 0;
 
-    DetermineShardToSendDataTo(my_cluster_num, my_shards_lo, my_shards_hi,
-                               shards, tmpCommittee, indexB2);
+    if (!shards.empty()) {
```

Comment:
Should we move the check earlier (e.g., start of this function)?

Reply:
if shards is empty then only send to lookup